### PR TITLE
Refactoring package mail

### PR DESCRIPTION
**Pull request by EnKarin**

### CODE REVIEW

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java [hunk 1]
```diff
@@ -38,57 +39,52 @@ public class PasswordResetController {
...
-        final ErrorListResponse response = new ErrorListResponse();
-        final boolean result = mailService.sendResetPassword(email);
-        if (result) {
-            return new ResponseEntity<>(HttpStatus.OK);
-        } else {
-            response.getErrors().add("user: Пользователь с таким email не найден");
-            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
-        }
+        mailService.sendResetPassword(email);
+        return ResponseEntity.status(HttpStatus.OK).build();
```
Comment (mfvanek) [inline src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java:51]:
```suggestion
        return ResponseEntity.ok();
```

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java [hunk 2]
```diff
@@ -38,57 +39,52 @@ public class PasswordResetController {
...
+            throw new PasswordsDontMatchException();
         }
+        resetPasswordService.updatePassword(token, userPasswordDto);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
+    }
+
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @ExceptionHandler(PasswordsDontMatchException.class)
+    public Map<String, String> passwordConflict(final PasswordsDontMatchException exc) {
+        return Collections.singletonMap("password", exc.getMessage());
```
Comment (mfvanek) [inline src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java:82]:
```suggestion
        return Map.of("password", exc.getMessage());
```

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java [hunk 3]
```diff
@@ -38,57 +39,52 @@ public class PasswordResetController {
...
+    }
+
+    @ResponseStatus(HttpStatus.FORBIDDEN)
+    @ExceptionHandler(TokenInvalidException.class)
+    public Map<String, String> tokenInvalid(final TokenInvalidException exc) {
+        return Collections.singletonMap("token", exc.getMessage());
```
Comment (mfvanek) [inline src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java:88]: Consider to restrict usage of obsolete Collections methods.

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-checkstyle-plugin</artifactId>
                <dependencies>
                    <dependency>
                        <groupId>com.puppycrawl.tools</groupId>
                        <artifactId>checkstyle</artifactId>
                        <version>${plugins.checkstyle.puppycrawl.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>com.thomasjensen.checkstyle.addons</groupId>
                        <artifactId>checkstyle-addons</artifactId>
                        <version>${plugins.checkstyle.thomasjensen.version}</version>
                    </dependency>
                </dependencies>

                                <module name="IllegalMethodCall">
                                    <property name="illegalMethodNames" value="emptyList, emptyMap, emptySet, emptyNavigableMap, emptyNavigableSet,
                                    emptySortedMap, emptySortedSet, singleton, singletonList, singletonMap"/>
                                </module>

**Review by mfvanek — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java (hunk 3 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java:88]: Maven refuses to find it

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java (hunk 1 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: Changed

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java (hunk 2 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: Changed

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java (hunk 3 above)
Comment (mfvanek) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: <dependency>
    <groupId>com.thomasjensen.checkstyle.addons</groupId>
    <artifactId>checkstyle-addons</artifactId>
    <version>7.0.1</version>
</dependency>

**Review by mfvanek — CHANGES_REQUESTED** [review]
I insist on using checkstyle "IllegalMethodCall"

File: src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java (hunk 3 above)
Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: Dependency 'com.thomasjensen.checkstyle.addons:checkstyle-addons:7.0.1' not found

Comment (mfvanek) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: Just update all deps
```
                <dependencies>
                    <dependency>
                        <groupId>com.puppycrawl.tools</groupId>
                        <artifactId>checkstyle</artifactId>
                        <version>10.3.1</version>
                    </dependency>
                    <dependency>
                        <groupId>com.thomasjensen.checkstyle.addons</groupId>
                        <artifactId>checkstyle-addons</artifactId>
                        <version>7.0.1</version>
                    </dependency>
                </dependencies>
```

Comment (EnKarin) [reply src/main/java/io/github/enkarin/bookcrossing/user/controllers/PasswordResetController.java]: Added

**Review by mfvanek — APPROVED** [review]

**Pull request merged**